### ISSUE TITLE
Support children HTMLCollection.

### ISSUE THREAD

**Issue by ilyavf** [canjs/can-simple-dom#36]
This is a bit related to #20, but still would be nice to have [children](https://developer.mozilla.org/en-US/docs/Web/API/ParentNode/children) as HTMLCollection.

**Comment by ilyavf** [comment]
Actually... just found @justinbmeyer's [comment](https://github.com/canjs/can-simple-dom/pull/16/files#r44734172) with a concern about performance.

May be it would be better to just define a property `children` with a getter, so that only who needs it will suffer. Smth like:

```
Object.defineProperty(Node.prototype, "children", {
    get: function () {
        return getChildren(this);
    }
});
```

This works for all the tests I created in the above PR, but I m not sure what you guys prefer @justinbmeyer @matthewp .

**Comment by matthewp** [comment]
If you use `el.childNodes.item(0)` there is no performance issues because childNodes is not an array-like, it's just a special type. You only suffer when you do `el.childNodes.length` because when you get the length you loop through the entire thing.  So I would prefer that `.children` work like `.childNodes`.

**Comment by ilyavf** [comment]
@matthewp I meant that in the 1st PR when for example you `appendChild(doc_frag)` it will call [getChildren](https://github.com/canjs/can-simple-dom/commit/151a1cbb275e0b064ac7d410722828df88e5c2bc#diff-6618e55302e5f1ee422b2299815d4fc4R188) which is an extra O(n).

**Comment by matthewp** [comment]
Yeah, I can see that, we definitely don't want to incur that cost.

**Comment by justinbmeyer** [comment]
I think this is something we can support once proxys are in all browsers.
